libcommon/strjoin.c

```diff
@@ -15,16 +15,16 @@ char *strjoin(const char *elems[], char *sep) {
  	size_t en = 0,
 	       i = 0,
 	       joinlen = 0;
-	char **elem = NULL,
-	     *joinbuf = NULL,
+	const char **elem = NULL;
+	char *joinbuf = NULL,
 	     *joinbufp = NULL;
 
-	for (elem = (char **)elems; *elem; elem++, en++)
+	for (elem = elems; *elem; elem++, en++)
 		joinlen += strlen(*elem);
 	joinlen += ((strlen(sep) * (en - 1)) + 1);
 
 	/* Populate the buffer. */
-	joinbuf = calloc(joinlen, sizeof(char));
+	joinbuf = malloc((joinlen * sizeof(char)));
```

Comment:
Why `malloc(( ))`

Reply:
Force of habit. I often enclosure mathematical expressions when using as input.

Reply:
Don't

Reply:
Ok, I will change this.